junit-test-result-parser/src/main/java/org/arquillian/smart/testing/parser/junit/JUnitTestResultParser.java

```diff
@@ -0,0 +1,99 @@
+package org.arquillian.smart.testing.parser.junit;
+
+import java.io.InputStream;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.Attribute;
+import javax.xml.stream.events.EndElement;
+import javax.xml.stream.events.StartElement;
+import javax.xml.stream.events.XMLEvent;
+import org.arquillian.smart.testing.spi.TestResultParser;
+import org.arquillian.smart.testing.spi.TestResult;
+
+public class JUnitTestResultParser implements TestResultParser {
+
+    @Override
+    public Set<TestResult> parse(InputStream junitInputStream) {
```

Comment:
I don't think so, I think that in most APIs that I have used they let callers close the stream.